### PR TITLE
Fix some ameridian-related harddels

### PR DESCRIPTION
**Pull request by r4d6**
## About The Pull Request
Fix some ameridian-related harddels.

### CODE REVIEW

File: code/modules/ameridian/ameridian_crystal.dm [hunk 1]
```diff
@@ -39,6 +39,7 @@
 /obj/structure/ameridian_crystal/Destroy()
 	..()
 	STOP_PROCESSING(SSobj, src)
+	golem?.node = null
```
Comment (nikothedude) [inline code/modules/ameridian/ameridian_crystal.dm:43]:
```suggestion
golem = null

```

for sanity's sake

File: code/modules/mob/living/carbon/superior_animal/ameridian/ameridian_golem.dm [hunk 2]
```diff
@@ -52,9 +52,12 @@
 	if(drop_amount)
 		var/obj/item/stack/material/ameridian/loot = new /obj/item/stack/material/ameridian(get_turf(src))
 		loot.amount = drop_amount
-	node?.golem = null // Remove the golem from the node since for some reason it doesn't do it with qdel()
 	qdel(src)
 
+/mob/living/carbon/superior_animal/ameridian_golem/Destroy()
+	..()
+	node?.golem = null
+
```
Comment (nikothedude) [inline code/modules/mob/living/carbon/superior_animal/ameridian/ameridian_golem.dm:60]:
```suggestion
node.null

```
sanity

**Comment by cdb-is-not-good** [comment]
tested 'n ready 2 go?

**Comment by r4d6** [comment]
> tested 'n ready 2 go?

Yes it is.

**Pull request merged**